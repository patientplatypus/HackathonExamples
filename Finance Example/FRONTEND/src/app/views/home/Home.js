// @flow weak

import React, {
  PureComponent
}                     from 'react';
import PropTypes      from 'prop-types';
import AnimatedView   from '../../components/animatedView/AnimatedView';
import {
  Button,
  Icon
}     from 'antd';

class Home extends PureComponent {
  static propTypes= {
    // react-router 4:
    match:    PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    history:  PropTypes.object.isRequired,
    
    // views:
    currentView:  PropTypes.string.isRequired,
    enterHome:    PropTypes.func.isRequired,
    leaveHome:    PropTypes.func.isRequired
  };

  componentDidMount() {
    const { enterHome } = this.props;
    enterHome();
  }

  componentWillUnmount() {
    const { leaveHome } = this.props;
    leaveHome();
  }

  render() {
    return(
      <AnimatedView>
        <div>
          <h1>
            ReactJS + Redux + Ant Design
          </h1>
          <h2>
            with Hot Reload!!!
          </h2>
          <h2>
            and React Router v4
          </h2>
          <h2>
            and webpack 3.x
          </h2>
          <h1>
            Starter
          </h1>
          <p>
            <Button
              type="primary"
              onClick={this.handlesOnGoAbout}
            >
              <Icon type="info-circle-o" />
              go to about
            </Button>
          </p>
        </div>
      </AnimatedView>
    );
  }

  handlesOnGoAbout = (
    event: SyntheticEvent<>
  ): void => {
    if (event) {
      event.preventDefault();
      const { history } = this.props;

      history.push('/about');
    }
  }
}

export default Home;
