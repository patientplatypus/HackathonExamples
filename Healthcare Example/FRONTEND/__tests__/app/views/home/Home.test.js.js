// @flow weak

import React            from 'react';
import Home             from '../../../../src/app/views/home/Home';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { 
  MemoryRouter
}                       from 'react-router';
import {
  shallow,
  mount
}                       from 'enzyme';

describe('Home component', () => {
  it('renders as expected', () => {
    const mockProps = {
      // react-router 4:
      match: {},
      location: {},
      history: {},
      // views:
      enterHome: () => {},
      leaveHome: () => {},
      currentView: 'About'
    };

    const component = renderer.create(
      <div>
        <MemoryRouter>
          <Home {...mockProps} />
        </MemoryRouter>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('triggers enterHome on mount', () => {
    const mockEnterHome = jest.fn();
    const mockProps = {
      // react-router 4:
      match: {},
      location: {},
      history: {},
      // views:
      enterHome: mockEnterHome,
      currentView: 'Home',
      leaveHome: () => {}
    };

    /* eslint-disable no-unused-vars */
    // shallow would not call componentDidMount, so we use mount is that case:
    const wrapper = mount(
      <Home {...mockProps} />
    );

    expect(mockEnterHome.mock.calls.length).toBe(1);
  });

  it('triggers leaveHome on unMount', () => {
    const mockLeaveHome = jest.fn();
    const mockProps = {
      // react-router 4:
      match: {},
      location: {},
      history: {},
      // views:
      enterHome: () => {},
      currentView: 'Home',
      leaveHome: mockLeaveHome
    };

    const wrapper = shallow(
      <Home {...mockProps} />
    );

    wrapper.unmount();

    expect(mockLeaveHome.mock.calls.length).toBe(1);
  });
});
