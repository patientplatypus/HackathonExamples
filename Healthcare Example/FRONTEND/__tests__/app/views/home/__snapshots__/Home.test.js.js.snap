// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <div>
      <h1>
        ReactJS + Redux + Ant Design
      </h1>
      <h2>
        with Hot Reload!!!
      </h2>
      <h2>
        and React Router v4
      </h2>
      <h2>
        and webpack 3.x
      </h2>
      <h1>
        Starter
      </h1>
      <p>
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <i
            className="anticon anticon-info-circle-o"
          />
          <span>
            go to about
          </span>
        </button>
      </p>
    </div>
  </section>
</div>
`;
